src/tests/page_heap_test.cc

```diff
@@ -64,7 +64,35 @@ static void TestPageHeap_Stats() {
   delete ph;
 }
 
+// Allocates all the last-level page tables we will need. Doing this before
+// calculating the base heap usage is necessary, because otherwise if any of
+// these are allocated during the main test it will throw the heap usage
+// calculations off and cause the test to fail.
+static void AllocateAllPageTables() {
+  // Make a separate PageHeap from the main test so the test can start without
+  // any pages in the lists.
+  tcmalloc::PageHeap* ph = new tcmalloc::PageHeap();
```

Comment:
unique_ptr ?

Reply:
Done

I updated the other uses in this file too, to keep them the same.